pkg/collector/nakadi_collector.go

```diff
@@ -72,14 +71,39 @@ func NewNakadiCollector(_ context.Context, nakadi nakadi.Nakadi, hpa *autoscalin
 		return nil, fmt.Errorf("metric-type must be either '%s' or '%s', was '%s'", nakadiMetricTypeConsumerLagSeconds, nakadiMetricTypeUnconsumedEvents, metricType)
 	}
 
+	// Either subscription-id or filtering via owning-application,
+	// event-types, and consumer-group is supported. If all are defined
+	// then subscription-id is used.
+	subscriptionFilter := &nakadi.SubscriptionFilter{}
+	if subscriptionID, ok := config.Config[nakadiSubscriptionIDKey]; ok {
+		subscriptionFilter.SubscriptionID = subscriptionID
+	}
+
+	if owningApplication, ok := config.Config[nakadiOwningApplicationKey]; ok {
+		subscriptionFilter.OwningApplication = owningApplication
+	}
+
+	if nakadiEventTypes, ok := config.Config[nakadiEventTypesKey]; ok {
+		eventTypes := strings.Split(nakadiEventTypes, ",")
+		subscriptionFilter.EventTypes = eventTypes
+	}
+
+	if consumerGroup, ok := config.Config[nakadiConsumerGroupKey]; ok {
+		subscriptionFilter.ConsumerGroup = consumerGroup
+	}
+
+	if subscriptionFilter.SubscriptionID == "" && (subscriptionFilter.OwningApplication == "" && len(subscriptionFilter.EventTypes) == 0 && subscriptionFilter.ConsumerGroup == "") {
+		return nil, fmt.Errorf("subscription-id or owning-application, event-types, and consumer-group must be specified on the metric")
```

Comment:
I think the error message can be improved a little with something like 
```golang
"either subscription-id or one of [owning-application, event-types, consumer-group] must be specified on the metric"
```
Just to make it clearer to the user what we expect from them.
Just to clar

Reply:
Updated